src/Data/Equivalence/STT.hs

```diff
@@ -405,3 +408,24 @@ removeClass eq v = do
         then return False
         else removeEntry (fromMaybe entry mentry)
              >> return True
+
+{-| This function returns all values represented by
+   some equivalence class. -}
+
+values :: (Monad m, Applicative m, Ord a) => Equiv s c a -> STT s m [a]
+values Equiv {entries = mref} = Map.keys <$> readSTRef mref
+
+{-| This function returns the list of
+   all equivalence classes. -}
+
+classes :: (Monad m, Applicative m, Ord a) => Equiv s c a -> STT s m [Class s c a]
+classes Equiv {entries = mref} = do
+  allEntries <- Map.elems <$> readSTRef mref
+  rootEntries <- filterM isRoot allEntries
+  mapM (fmap Class . newSTRef) $ rootEntries
```

Comment:
It seems that a law could be that `classes e` should be the same set as `map (getClass e) (values e)`.  However, since `newSTRef` is involved here, the exact meaning of _the same set_ needs to be figured out.